compiler/nim.nim

```diff
@@ -56,70 +58,43 @@ proc processCmdLine(pass: TCmdLinePass, cmd: string; config: ConfigRef) =
     if optRun notin config.globalOptions and config.arguments.len > 0 and config.command.normalize != "run":
       rawMessage(config, errGenerated, errArgsNeedRunOption)
 
+method mainCommand(self: ProgNim, graph: ModuleGraph)=
+  mainCommand(graph)
+
 proc handleCmdLine(cache: IdentCache; conf: ConfigRef) =
-  condsyms.initDefines(conf.symbols)
+  let self = ProgNim(name: "nim_compiler")
```

Comment:
NOTE: this PR adds
`defineSymbol conf.symbols, "nim_compiler"` 
for symmetry with what's done in numsuggest and nimfix